internal/kafka/internal/services/cloud_providers_test.go

```diff
@@ -501,3 +555,113 @@ func Test_ListCloudProviderRegions(t *testing.T) {
 		})
 	}
 }
+
+func Test_cloudProvidersService_ListCachedCloudProviderRegions(t *testing.T) {
+	type fields struct {
+		providerFactory   clusters.ProviderFactory
+		connectionFactory *db.ConnectionFactory
+		cache             *cache.Cache
+	}
+	type args struct {
+		id string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    []api.CloudRegion
+		setupFn func()
+		wantErr *Error.ServiceError
+	}{
+		{
+			name: "successful get cloud provider regions from various provider types",
+			fields: fields{
+				connectionFactory: db.NewMockConnectionFactory(nil),
+				providerFactory: &clusters.ProviderFactoryMock{
+					GetProviderFunc: func(providerType api.ClusterProviderType) (clusters.Provider, error) {
+						return &clusters.ProviderMock{
+							GetCloudProvidersFunc: func() (*types.CloudProviderInfoList, error) {
+								return &types.CloudProviderInfoList{
+									Items: []types.CloudProviderInfo{
+										{
+											ID:          "azure",
+											Name:        "azure",
+											DisplayName: "azure",
+										},
+										{
+											ID:          "aws",
+											Name:        "aws",
+											DisplayName: "aws",
+										},
+									},
+								}, nil
+							},
+							GetCloudProviderRegionsFunc: func(providerInf types.CloudProviderInfo) (*types.CloudProviderRegionInfoList, error) {
+								return &types.CloudProviderRegionInfoList{
+									Items: []types.CloudProviderRegionInfo{
+										{
+											ID:              "af-east-1",
+											CloudProviderID: providerInf.ID,
+											Name:            "af-east-1",
+											DisplayName:     "af-east-1",
+										},
+										{
+											ID:              "af-south-1",
+											CloudProviderID: providerInf.ID,
+											Name:            "af-south-1",
+											DisplayName:     "af-south-1",
+										},
+									},
+								}, nil
+							},
+						}, nil
+					}},
+				cache: cache.New(5*time.Minute, 10*time.Minute),
+			},
+			setupFn: func() {
+				mocket.Catcher.Reset()
+				mocket.Catcher.NewMock().
+					WithQuery("SELECT DISTINCT").
+					WithReply([]map[string]interface{}{{"provider_type": "ocm"}, {"provider_type": "standalone"}})
```

Comment:
As last match, always add
```go
mocket.Catcher.NewMock().WithExecException().WithQueryException()
```
This way if the query is not matched you will get an error (otherwise it will silently exit)